### ISSUE TITLE
Android: Emulator Deadlock with ControlX-modified boot1

### ISSUE THREAD

**Issue by MystiK791** [nspire-emus/firebird#276]
I have a really strange issue. I have a TI-Nspire CX that I installed CAS OS on with ControlX. I built the flash on Windows with a CX CAS OS image. When I try to run it on Android, however, the emulator hangs on a black screen after completing the initial boot process. No button presses are responsive. Is there anything I'm doing wrong here?
![Screenshot_20221104-195333.png](https://user-images.githubusercontent.com/69944483/200095923-10328062-c4bb-4feb-af49-a72311a739da.png)
![Screenshot_20221104-195328.png](https://user-images.githubusercontent.com/69944483/200095927-969ef879-8b96-4385-85b4-75330dc606e3.png)

**Comment by adriweb** [comment]
But does it work fine on the computer?

**Comment by MystiK791** [comment]
No, see for yourself:
![image](https://user-images.githubusercontent.com/69944483/200128752-0e8c3b7e-7679-455f-b828-1d86689f192f.png)
![image](https://user-images.githubusercontent.com/69944483/200128766-e1a81e4f-f42b-481e-bf99-fc00abe64b1d.png)

**Comment by MystiK791** [comment]
For reference, this is what the boot screen looks like on real hardware ([source](https://www.youtube.com/watch?v=6UU0qagcB8o)):
![image](https://user-images.githubusercontent.com/69944483/200128967-5043196f-7e95-4418-a11a-9762e3f83e6d.png)

**Comment by adriweb** [comment]
Seems like you didn't dump/use the manuf file from your calc ?

**Comment by Vogtinator** [comment]
Does the serial log show anything? Looks like the manuf got changed at least because the graphical splash components are missing, as expected with nBoot.

**Comment by MystiK791** [comment]
> Seems like you didn't dump/use the manuf file from your calc ?

I did

**Comment by MystiK791** [comment]
> Does the serial log show anything? Looks like the manuf got changed at least because the graphical splash components are missing, as expected with nBoot.

I'll have to check later.

**Comment by MystiK791** [comment]
![image](https://user-images.githubusercontent.com/69944483/200136930-b966d717-fd85-4a8a-b028-272d8036bdcb.png)
There's the issue.

**Comment by Vogtinator** [comment]
Right, that explains it. nBoot resp. ControlX work by booting from the diags partition, which you need to select explicitly in firebird by using the `Boot Diags with Kit` option. I don't think that's reachable in the Mobile UI though...

**Comment by MystiK791** [comment]
Alright, so booting from diags gets me a little farther on PC. I'm currently waiting to see if I can get past this screen:
![image](https://user-images.githubusercontent.com/69944483/200180923-9505c118-42b9-439b-94e9-acfefa3835a6.png)

**Comment by MystiK791** [comment]
Thanks for being so helpful so far 😁

I have a vanilla Boot1 and Flash image I made before I installed ControlX, I'll use those on my phone for now.
